subprojects/core/src/test/groovy/org/gradle/util/internal/DistributionLocatorIntegrationTest.groovy

```diff
@@ -46,12 +48,26 @@ class DistributionLocatorIntegrationTest extends Specification {
         urlExist(locator.getDistributionFor(distributions.mostRecentReleaseSnapshot.version))
     }
 
-    void urlExist(URI url) {
+    private void urlExist(URI url) {
+        int responseCode = 0
+        for (int attempt = 1; attempt <= NUM_RETRIES; attempt++) {
```

Comment:
I would expect we already have a utility somewhere for retries/backoff. It's pretty simple here so I don't really have a problem duplicating but if it's easy to find an existing fixture id rather we use that

Reply:
We have `ExponentialBackoff`.

Reply:
Good idea, thanks for pointing that out.

I've reworked the test to use exponential backoff and clean it up some.  Please take another look.